File: common/src/main/java/juuxel/adorn/menu/DrawerMenu.java
package juuxel.adorn.menu;

import net.minecraft.world.entity.player.Inventory;
import net.minecraft.world.Container;
import net.minecraft.world.SimpleContainer;
import net.minecraft.world.inventory.ContainerLevelAccess;
import net.minecraft.core.BlockPos;

public final class DrawerMenu extends SimpleMenu {
    private static final int WIDTH = 5;
    private static final int HEIGHT = 3;

    public DrawerMenu(int syncId, Inventory playerInventory, Container inventory, ContainerLevelAccess context) {
        super(AdornMenus.DRAWER.get(), syncId, WIDTH, HEIGHT, inventory, playerInventory, context);
    }

    public static DrawerMenu load(int syncId, Inventory playerInventory, BlockPos pos) {
        var context = ContainerLevelAccess.create(playerInventory.player.level(), pos);
        return new DrawerMenu(syncId, playerInventory, new SimpleContainer(WIDTH * HEIGHT), context);
    }
}

File: common/src/main/kotlin/juuxel/adorn/menu/DrawerMenu.kt
package juuxel.adorn.menu

import net.minecraft.entity.player.PlayerInventory
import net.minecraft.inventory.Inventory
import net.minecraft.inventory.SimpleInventory
import net.minecraft.menu.MenuContext
import net.minecraft.network.PacketByteBuf

class DrawerMenu(
    syncId: Int,
    playerInventory: PlayerInventory,
    container: Inventory,
    context: MenuContext
) : PalettedMenu(AdornMenus.DRAWER, syncId, DIMENSIONS, container, playerInventory, context) {
    companion object {
        private val DIMENSIONS = 5 by 3

        fun load(syncId: Int, playerInventory: PlayerInventory, buf: PacketByteBuf): DrawerMenu {
            val pos = buf.readBlockPos()
            val context = MenuContext.create(playerInventory.player.world, pos)
            return DrawerMenu(syncId, playerInventory, SimpleInventory(DIMENSIONS.size), context)
        }
    }
}
